modules/oe_subscriptions_anonymous/oe_subscriptions_anonymous.tokens.inc

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * @file
+ * Token callbacks for the Anonymous subscriptions module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\Render\BubbleableMetadata;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_token_info().
+ */
+function oe_subscriptions_anonymous_token_info(): array {
+  $type = [
```

Comment:
All the token approach is wrong.
Tokens works off some data, eg `node:field_something`, `user:cancel-url`, and they are recursively generated. E.g. given there's a message as context, and the message has a user property, you get `[message:author:cancel-url]`. 
Below we check if there's message as data, and in that case we add the token in the type.
The correct solution was to just add the `subscriptions-page-url` (note the dashes and not underscores) to the user type.
Furthermore, the token shouldn't handle a link, but just the url (as above). It might be needed in a plain text email, so there's no html available, or it might need further customisation in an HTML context.

Reply:
Understood, I was applying wrong logic, thanks